source/material_model/entropy_model.cc

```diff
@@ -173,14 +186,38 @@ namespace aspect
           const double entropy = in.composition[i][entropy_index];
           const double pressure = this->get_adiabatic_conditions().pressure(in.position[i]) / 1.e5;
 
-          out.densities[i] = entropy_reader[0]->density(entropy,pressure);
-          out.thermal_expansion_coefficients[i] = entropy_reader[0]->thermal_expansivity(entropy,pressure);
-          out.specific_heat[i] = entropy_reader[0]->specific_heat(entropy,pressure);
+          // Loop over all material files, and store the looked-up values for all compositions.
+          for (unsigned int j=0; j<material_file_names.size(); ++j)
+            {
+              eos_outputs.densities[j] = entropy_reader[j]->density(entropy, pressure);
+              eos_outputs.thermal_expansion_coefficients[j] = entropy_reader[j]->thermal_expansivity(entropy,pressure);
+              eos_outputs.specific_heat_capacities[j] = entropy_reader[j]->specific_heat(entropy,pressure);
+
+              const Tensor<1, 2> pressure_unit_vector({0.0, 1.0});
+              eos_outputs.compressibilities[j] = ((entropy_reader[j]->density_gradient(entropy,pressure)) * pressure_unit_vector) / eos_outputs.densities[j];
+            }
+
+          // Calculate volume fractions from mass fractions
+          // If there is only one lookup table, set the mass and volume fractions to 1
+          std::vector<double> mass_fractions;
+          if (material_file_names.size() == 1)
+            mass_fractions.push_back(1.0);
```

Comment:
This is not wrong, but it would be faster and I think also easier to understand if you declared the `mass_fractions` further up at the same spot where you declare the `volume_fractions`. You can then also initialize them with the correct size (`material_file_names.size()`) and change this code to:
```suggestion
          if (material_file_names.size() == 1)
            mass_fractions [0] = 1.0;
```